hermes/probe/delete_file.go

```diff
@@ -0,0 +1,104 @@
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"github.com/googleinterns/step224-2020/hermes"
+	"google.golang.org/api/iterator"
+
+	probepb "github.com/googleinterns/step224-2020/config/proto"
+	m "github.com/googleinterns/step224-2020/hermes/probe/metrics"
+)
+
+// deleteRandomFile deletes a random file in the target storage system bucket.
+// It then checks that the file has been deleted by trying to get the object.
+// Arguments:
+// TODO(evanSpendlove): Update comment here
+//	- ctx: pass the context so this probe can be cancelled if needed.
+//	- config: pass the HermesProbeDef config for the probe calling this function.
+//	- target: pass the target run information.
+//	- client: pass an initialised storage client for this target system.
+//	- logger: pass the logger associated with the probe calling this function.
+// Returns:
+//	- fileID: returns the ID of the file delete OR a missing file to be created if one is found.
+//	- err:
+//		- [...].hermes_file_missing: the file to be deleted does not exist in Hermes' StateJournal.
+//		- [...].file_missing: the file to be deleted could not be found in the target bucket.
+//		- [...].bucket_missing: the target bucket on this target system was not found.
+//		- [...].probe_failed: there was an error during one of the API calls and the probe failed.
+//		- [...].deleted_file_found: the file was deleted but it is still found in the target bucket.
+//		- [...].list_bucket_failed: the listBucket operation failed when checking if the target file was deleted.
+func DeleteRandomFile(ctx context.Context, config *probepb.HermesProbeDef, target *Target, client *stiface.Client, logger *logger.Logger) (int, error) {
+	bucket := target.Target.GetBucketName()
+
+	fileID := hermes.PickFileToDelete()
+
+	filename, ok := target.Journal.Filenames[int32(fileID)]
+	if !ok {
+		return fileID, fmt.Errorf("deleteRandomFile(id: %d).hermes_file_missing: could not delete file %s, file not found", fileID, filename)
+	}
+
+	file := (*client).Bucket(bucket).Object(filename)
+
+	start := time.Now()
+	if err := file.Delete(ctx); err != nil {
+		var status string
+		if err == storage.ErrObjectNotExist {
+			status = m.ExitStatuses[m.FILE_MISSING]
+		} else if err == storage.ErrBucketNotExist {
+			status = m.ExitStatuses[m.BUCKET_MISSING]
+		} else {
+			status = m.ExitStatuses[m.PROBE_FAILED]
+		}
+		target.LatencyMetrics.ApiCallLatency[m.ApiCalls[m.API_DELETE_FILE]][status].Metric("latency").AddFloat64(time.Now().Sub(start).Seconds())
+		return fileID, fmt.Errorf("deleteRandomFile(id: %d).%s: could not delete file %s: %w", fileID, status, filename, err)
+	}
+	target.LatencyMetrics.ApiCallLatency[m.ApiCalls[m.API_DELETE_FILE]][m.ExitStatuses[m.SUCCESS]].Metric("latency").AddFloat64(time.Now().Sub(start).Seconds())
+
+	query := &storage.Query{Prefix: filename}
+	start = time.Now()
+	it := (*client).Bucket(bucket).Objects(ctx, query)
```

Comment:
Please pick a more descriptive name than `it` - `it` is especially bad given how it's semantically meaningful in at least some languages.

Reply:
Ah apologies for this. `it` was used in the godoc for the storage library, so I kept it. I've changed this to `objects` as it is an iterator over objects.